src/bst.py

```diff
@@ -0,0 +1,329 @@
+"""Implementation of a Binary Search Tree."""
+
+from my_queue import Queue
+from stack import Stack
+
+
+class Node(object):
+    """Node class."""
+
+    def __init__(self, val):
+        """Node instantiation."""
+        self.val = val
+        self.leftChild = None
+        self.rightChild = None
+
+
+class BinarySearchTree(object):
+    """Binary Search Tree Class."""
+
+    def __init__(self):
+        """Binary Search Tree instantiation."""
+        self.root = None
+        self._size = 0
+
+    def insert(self, val):
+        """Insert value into Binary Search Tree."""
+        if self.root is None:
+            self.root = Node(val)
+            self._size = 1
+            return
+        node = Node(val)
+        trv = []
+        curr_node = self.root
+        while curr_node:
+            trv.append(curr_node)
+            if val > curr_node.val:
+                curr_node = curr_node.rightChild
+            else:
+                curr_node = curr_node.leftChild
+
+        if node.val > trv[-1].val:
+            trv[-1].rightChild = node
+        else:
+            trv[-1].leftChild = node
+        self._size += 1
+        trv.append(node)
+        return trv[::-1]
```

Comment:
Why include this return statement, you don't want to see your tree each time you insert a node.